fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImpl.java

```diff
@@ -267,14 +316,18 @@ public void postDruidQuery(
             timerName = DRUID_WEIGHTED_QUERY_TIMER + String.format(format, seqNum);
         }
 
+        BoundRequestBuilder requestBuilder = webClient.preparePost(serviceConfig.getUrl())
+                .setBody(entityBody)
+                .addHeader("Content-Type", "application/json");
+
+        anyHeaders.get().forEach((k, v) -> requestBuilder.addHeader(k, v));
+
```

Comment:
It does not appear that headers are maplike in that there can't be more than one of a given key value.  Given this, you might be better served with an iterable here.